### ISSUE TITLE
Use buttons in place of anchors when onClick is provided with no href.

### ISSUE THREAD

**Issue by mtscout6** [react-bootstrap/react-bootstrap#442]
For better accessibility compliance we should be using a `button` in place on and `anchor` for onClick handlers with no href. One of the best ways to accomplish this is probably to extract most of the logic in `Button` to a `NonStyledButton` which will only take the classes pushed to it. Then the actual `Button` could delegate to that. This may be the easiest way to replicate this logic throughout the library.

**Comment by AlexKVal** [comment]
It is already realized as
[src/Button.js#L42](https://github.com/react-bootstrap/react-bootstrap/blob/6ffa325871e1ba5f606cc513ec20def7e782bb86/src/Button.js#L42)

Anchor version has `role="button"` [src/Button.js#L59](https://github.com/react-bootstrap/react-bootstrap/blob/6ffa325871e1ba5f606cc513ec20def7e782bb86/src/Button.js#L59)
and `button` is just a `button` [src/Button.js#L66](https://github.com/react-bootstrap/react-bootstrap/blob/6ffa325871e1ba5f606cc513ec20def7e782bb86/src/Button.js#L66)
